### PR TITLE
[7.x] Add config options for Server/Agent communication via TLS (#2224)

### PR DESCRIPTION
**Pull request by graphaelli**
Backports the following commits to 7.x:
 - Add config options for Server/Agent communication via TLS (#2224)
 - Make client_auth optional by default. (#2281)

depends on elastic/apm-server#2326 to fix tests

### CODE REVIEW

File: changelogs/head.asciidoc [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+== APM Server version HEAD
+
+[float]
+==== Added
+- Support more SSL config options for agent/server communication {pull}2224[2224].
+
+[float]
+==== Removed
+
+https://github.com/elastic/apm-server/compare/7.2\...master[View commits]
```
Comment (simitt) [inline changelogs/head.asciidoc]: The changelog entry should probably go into `7.2` and not `head`.

File: beater/config_test.go [hunk 2]
```diff
@@ -84,7 +87,11 @@ func TestConfig(t *testing.T) {
 				WriteTimeout:    4000000000,
 				ShutdownTimeout: 9000000000,
 				SecretToken:     "1234random",
-				SSL:             &SSLConfig{Enabled: &truthy, Certificate: outputs.CertificateConfig{Certificate: "1234cert", Key: "1234key"}},
+				TLS: &tlscommon.ServerConfig{
+					Enabled:     &truthy,
+					CAs:         []string{"./ca.cert.pem"},
+					Certificate: outputs.CertificateConfig{Certificate: "1234cert", Key: "1234key"},
+					ClientAuth:  4}, //4=RequireAndVerifyClientCert
```
Comment (simitt) [inline beater/config_test.go]: This needs to be changed to `0`, as libbeat changed the default back to `none`.

File: changelogs/head.asciidoc (hunk 1 above)
Comment (jalvz) [reply changelogs/head.asciidoc]: fwiw, on my duped version of this pr, i chosen to remove the head entry in the conflict resolution, with the intention of doing the usual "fix changelogs" PR later on

Comment (graphaelli) [reply changelogs/head.asciidoc]: moved to 7.2

File: beater/config_test.go (hunk 2 above)
Comment (graphaelli) [reply beater/config_test.go]: libbeat now updated

Comment (simitt) [reply beater/config_test.go]: 7.x is not on the latest libbeat version, that's why `ClientAuth: 4` is ok here. Once we update to the latest, it should be 0 again.

**Pull request merged**